### ISSUE TITLE
Discussion: User feedback about Replies UX

### ISSUE THREAD

**Issue by BeccaMelhuish** [citizenos/citizenos-fe#660]
@kevincrepin @ilmartyrk @anettlinno 

In user research on setting up public discussions, the user gave feedback about the replies functionality not being good UX. He would like to use the platform for a participatory budgeting ideation process, but this is one thing he flagged as a big downside. He would still consider using the platform as it is, but only because he doesn't expect to get more than a few people engaging in the topic (but if it went well, and he had more, he would see it as a real issue for him. 

His issue was that all the replies are in a straight line and in chronological order, so it's not possible to see which reply someone is replying to. (His third reply was a reply to the first reply, but was placed at the bottom).  

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/8b5b3c70-a271-43d5-95a8-db592bc56b1c)

Some excerpts from the test on this bit: 

- “It doesn’t identify which response I’m responding to”
- “This would be quite important. Cause if you have different discussions it can get quite mixed up. And if somebody posted just a minute before you, you didn’t even know that your response was the third because you were typing it for a long time, and then nobody even remembers what I’m responding to. “
- (I asked if it would help that it shows different names, as he was replying to himself). “If the same person has said 5 things, and I’m saying ‘no this is not true at all, you’re lying about it’ - which one? Maybe he’s lying about something but not all.”   
- “When you look at Facebook discussions you can see OK this is a response to something. And without that, yeah that’s inconvenient. Responding to a response should be easier, and without that - that’s a major downside to me.” 

So a proposal from both myself and the user, would be that we try to follow as closely as possible how Facebook discussions work. They work very smoothly, with often hundreds of replies, and have presumably been heavily UX tested. They also work well both on desktop and mobile. Could we do the same? 

Facebook has several 'layers' of indentation, and also these little lines showing to which comment someone is replying:  

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/7f243cb3-f0b3-4063-b2ad-3f4d1b9b8339)

My personal feeling is that if discussions is one of our two headline features ('a participation platform for discussions and voting') then we should make sure the discussion part itself works really well :) 

It was previously discussed in [this issue](https://github.com/citizenos/citizenos-fe/issues/478) then we decided to test with users first. 

(This feedback is only based on one user, but the rest of the tests so far haven't tested long strings of replies - he just tried it out himself as it was an important consideration for him to use the platform. We might get more insights as the tests go on, but I think it's quite clear it's needed already?).

**Comment by BeccaMelhuish** [comment]
As it's such an important one I'd say this should ideally be fixed in Phase I as quite a high priority. In an ideal world, before we test on participants, which is planned for next week (second half of the week most likely), which I assume isn't realistic! Have added 'pre-UX tests' label, but will give a more realistic date to go with that once I know if it'll go ahead and how much time would take to implement :)

**Comment by kevincrepin** [comment]
I don't fully understand the reference to fb replies, because as far as I can see (but I don't have FB myself, so pls let me know if I'm completely wrong here) , they only go only one level deeper than us, so basically only 2 levels under a (pro/con/neutral) message.

As mentioned if you put no limit cap on this it will anyways break easily, which is probably why other platforms don't allow this and as far as I can see FB also flattens at the 3rd level. I'd love an example of this "It doesn’t identify which response I’m responding to" at the 3rd level of FB, because I think they also do chronological + include the person's name who you're replying to. Also keep in mind this is kinda how non-direct messaging works. If you spend a lot of time writing something, other people might get their response in before you. And if the person you're replying to is engaged in another conversation at the same level there is no way to distinguish this without unlimited levels...


Example to show FB also will put your reply at "the end" even if you're replying to someone "in the middle":
<img width="647" alt="Screenshot 2024-01-21 at 16 27 27" src="https://github.com/citizenos/citizenos-fe/assets/26792281/0397777b-4a44-4409-ba02-103858cd4112">



So if we want to do it like FB, what can we do? We can have one more level, which probably still would work ok. And we can have the connecting lines as they have, but since we don't have the user's profile photo out of the comment box, it loses a bit of it's purpose in my opinion...

https://www.figma.com/file/gOJ4i00538e0qaP6IKIPVE/Platform-2.0-UI?type=design&node-id=11460%3A83032&mode=design&t=cdGvUcp5EzmNWfSd-1

![Desktop](https://github.com/citizenos/citizenos-fe/assets/26792281/7ae454a6-85d9-4967-bca5-3c9e3b85c0ee)

**Comment by anettlinno** [comment]
@kevincrepin what you proposed makes the discussion much more traceable. Keeping it chronological and including person´s name you are responding to, makes it very clear! IMO 2 level under the main argument is all we need :) 

I support implementing this 💯

**Comment by BeccaMelhuish** [comment]
One more level deep plus the lines gets 💯 from me too! 

It's true even Facebook's doesn't work as well as the user was suggesting, but I think these two things will help a lot :)

**Comment by anettlinno** [comment]
Triage 98. Estimate dev time 8 hours. Sending it to do.

**Comment by ilmartyrk** [comment]
@anettlinno @kevincrepin @BeccaMelhuish  It was quite a pain to create, I even tried to solve the flattening from second level replies with ChatGPT, but it couldn't solve it. Finally managed to get a proper solution, hope it works and suits better

**Comment by BeccaMelhuish** [comment]
Tried it out and this gives me so much joy to see! :D :D 

It's not totally perfect in that sometimes when you reply to one of the replies to a reply, the line doesn't show to which one you replied, but it seems that's how Facebook does it also, and it would get rather messy-looking and hard to follow if it did. 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/e2518716-e33e-487a-9aeb-2728dcf992a6)

The only problem we do have is that it only shows 2 replies when there are 6 (only counting first level, when should count all) :) 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/a9e2ee49-5baf-4665-8276-b9ffeb47a807)

**Comment by ilmartyrk** [comment]
@BeccaMelhuish fixed the replies count

**Comment by BeccaMelhuish** [comment]
Tested and now closing :)
